server.js

```diff
@@ -0,0 +1,309 @@
+'use strict';
+
+const http = require('http');
+const fs = require('fs');
+const querystring = require('querystring');
+
+const PORT = process.env.PORT || 8080;
+
+const Users = {
+  gene: 'bean'
+};
+
+const server = http.createServer((req, res) => {
+  if (req.method === 'GET') {
+    let uri;
+
+    if ((req.url = '/')) {
+      uri = '/index.html';
+    } else {
+      uri = req.url;
+    }
+
+    fs.readFile('./public' + uri, 'utf8', (err, data) => {
+      if (err) {
+        if (err.code === 'ENOENT') {
+          res.writeHead(404);
+          fs.readFile('./public/404.html', 'utf8', (err, data) => {
+            if (err) {
```

Comment:
You should turn this into a guard by just returning the `throw err`, then the `res.end(data);` doesn't need to be in an else.